src/Checkbox.js

```diff
@@ -1,69 +1,52 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 
-class Checkbox extends Component {
-  constructor(props) {
-    super(props);
-  }
+const Checkbox = ({
+  id,
+  className,
+  indeterminate,
+  filledIn,
+  label,
+  onChange,
+  ...props
+}) => {
+  const [checked, setChecked] = useState(props.checked);
```

Comment:
let's deconstruct checked, and that way we dont need to delete on L25

Reply:
also, what happens if we pass checked as a new prop to this component, will it update?

Reply:
> let's deconstruct checked, and that way we dont need to delete on L25

I've done this by purpose. Otherwise we have to change the `checked` of `useState` to another, less semantic, name.

What do you suggest?

Reply:
defaultChecked?

Reply:
Well, I thought about `defaultChecked`, the thing is that it's misleading. 
We use it to track the `current` state of the `checkbox`, not the `default` one, as this name might suggest. :thinking: 
What do you think?

Reply:
> also, what happens if we pass checked as a new prop to this component, will it update?

Good point, I missed this one! :man_facepalming: 
It seems we have to `useEffect` here! :smile: